site/app/libraries/FileUtils.php

```diff
@@ -767,4 +767,37 @@ public static function getTopEmptyDir(string $path, string $base_path, &$results
             $results[] = $parent_dir;
         }
     }
+
+
+    /**
+     * Given a path, determines if there are characters in it that could lead to a security issue,
+     * if not return true
+     *
+     * @param string $path Absolute path to file or directory
+     * @return bool
+     */
+    public static function validPath($path) {
+        $disallowed = [
+            ':',    // Drive separator (Windows)
+            '*',    // Wildcard character
+            '?',    // Wildcard character
+            '"',    // Double quote
+            '<',    // Less than
+            '>',    // Greater than
+            '|',    // Pipe
+            '\0'    // Null character
+        ];
+
+        foreach ($disallowed as $char) {
+            if (strpos($path, $char) !== false) {
+                return false;
+            }
+        }
+
+        if (substr($path, 0, 2) === '..' || substr($path, -2) === '..') {
+            return false;
+        }
```

Comment:
Is this just checking if the first or last 2 characters are `..`? A better check you can do is too iterate over all of the directory names and see if they equal `..`. You can just split by `/`.